File: src/main/java/com/fatec/destino/model/pacote/tag/Tag.java
package com.fatec.destino.model.pacote.tag;

import com.fatec.destino.model.pacote.Pacote;
import jakarta.persistence.*;
import lombok.*;

import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "TAG_TAGs")
public class Tag {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "TAG_ID", nullable = false)
    private long id;

    @Column(name = "TAG_NOME", nullable = false)
    private String nome;

    // Relações

    @ManyToMany(mappedBy = "tags")
    private Set<Pacote> pacotes;

}

File: src/main/kotlin/com/fatec/destino/model/pacote/tag/Tag.kt
package com.fatec.destino.model.pacote.tag

import com.fatec.destino.model.pacote.Pacote
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.ManyToMany
import jakarta.persistence.Table

@Entity
@Table(name = "tag_tag")
class Tag(
    @Column(name = "TAG_NOME", length = 50, nullable = false, unique = true)
    var nome: String
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "TAG_ID", nullable = false)
    var id: Long? = null

    @ManyToMany(mappedBy = "tags")
    var pacotes: Set<Pacote> = emptySet()
}
